harness/determined/tensorboard/hdfs.py

```diff
@@ -31,13 +31,13 @@ def __init__(
         self.client.makedirs(str(self.sync_path))
 
     @util.preserve_random_state
```

Comment:
hm, these decorators are going to affect global state in a not-nice way when we start running the sync in the background.

Probably better to just delete them.  We are generally not as concerned about bit-perfect reproducibility as we were when I added these decorators.

Reply:
Hm, optionally we could put the preserve_random_state behavior at the base `sync()` layer, only in the non-async case.

Reply:
SharedFSTensorboardManager's `_sync_impl` has no @util.preserve_random_state decorator, therefore, we cannot uniformly wrap _sync_impl if async is true with util.preserve_random_state for all subclasses of TensorboardManager. 

For now, I am deleting the decorators. 

If we really want, I think we can wrap the `self._sync_impl `function with `util.preserve_random_state`  in the subclasses' `__init__ ` function that make use of @util.preserve_random_state decorator currently, conditional on the async_upload flag. But if this does not matter much, then deleting would make the code simpler.

Thoughts?

Reply:
> SharedFSTensorboardManager's _sync_impl has no @util.preserve_random_state decorator, therefore, we cannot uniformly wrap _sync_impl if async is true with util.preserve_random_state for all subclasses of TensorboardManager.

I actually think preserve_random_state would have no effect on shared_fs, otherwise it would already have the decorator.

The original purpose of preserve_random_state was to ensure that backoff-with-retry behavior of the cloud upload APIs didn't affect any rng state that could affect the output of training.  The reason we cared about that was that we used to advertise "reproducibility" as one of our biggest features, and we even would say things like, "if you use us and train on a CPU we can offer bit-perfect reproducibility".

When we were mature enough to actually attract customers, bit-perfect reproducibility was definitely not interesting to anybody, much less limiting training to CPUs.

But still, I think protecting RNG state is a good idea when we can afford to do it.  I think you might need to add a variation of `util.preserve_random_state` that you could use like this:
```python
if not run_async:
    with util.preserve_random_state_contextmgr():
        _sync_impl(...)
```

That is a very small increase in code complexity I think.

Reply:
Got it, I will make the update then!